beacon-chain/core/peerdas/helpers.go

```diff
@@ -675,3 +675,12 @@ func RecoverCellsAndProofs(
 
 	return recoveredCellsAndProofs, nil
 }
+
+func CellsCount(cellsAndProofs []kzg.CellsAndProofs) int {
```

Comment:
This function is only used in `reconstructDataColumns`. Could you move it in the corresponding file and make it private?

Also, the `range` expression you use make a copy of the cells and proofs, which are quite heavy.
https://go.dev/tour/moretypes/16

It would be better to use the `i++` version of the loop which does not copy.